nutils/topology.py

```diff
@@ -845,7 +833,16 @@ def interfaces(self):
 
   @property
   def points(self):
-    return UnstructuredTopology(itertools.chain.from_iterable(ptopo.references for ptopo in self.pgroups.values()), itertools.chain.from_iterable(ptopo.transforms for ptopo in self.pgroups.values()), itertools.chain.from_iterable(ptopo.opposites for ptopo in self.pgroups.values()), ndims=0).withgroups(self.pgroups)
+    ptopos = []
+    pnames = []
+    topo = self
+    while isinstance(topo, WithGroupsTopology):
+      for pname, ptopo in topo.pgroups.items():
+        if pname not in pnames:
+          pnames.append(pname)
+          ptopos.append(ptopo)
+      topo = topo.basetopo
+    return DisjointUnionTopology(ptopos, pnames)
```

Comment:
Are the pgroups/ptopos always disjoint?

Reply:
In changing to UnionTopology I must have pushed a wrong commit causing github to close the PR. Opening a new one.